cmd/app.go

```diff
@@ -326,3 +352,41 @@ func lastReadJournalTimeStamp() uint64 {
 
 	return lastRead
 }
+
+func handleAuditLogMetrics(
+	ctx context.Context,
+	eg *errgroup.Group,
+	pprov *metrics.PrometheusMetricsProvider,
+	auditMetricsSecondsInterval int,
+	auditLogWriteTimeSecondThreshold int,
+	enableAuditMetrics bool,
+) {
+	if !enableAuditMetrics {
+		return
+	}
+
+	auditLogFilePath := "/var/log/audit/audit.log"
+
+	eg.Go(func() error {
+		tickChan := time.NewTicker(time.Second * time.Duration(auditMetricsSecondsInterval)).C
+		for {
+			select {
+			case <-tickChan:
+				s, err := os.Stat(auditLogFilePath)
+				if err != nil {
+					return fmt.Errorf("error stat-ing %s", auditLogFilePath)
```

Comment:
I think you will want to `continue` (and maybe write a log message) here if an error occurs. A transient stat failure here will result in the program exiting because we return to `eg.Go()`.

Reply:
ah ok. ill change to continue

Reply:
done